cime_config/hist_config.py

```diff
@@ -0,0 +1,1126 @@
+        pobj.add_syntax_err(emsg)
+        return False
+
+    @property
+    def max_frames(self):
+        """Return the max_frames property for this HistoryVolConfig object"""
+        return self.__max_frames
+
+    @property
+    def write_nstep0(self):
+        """Return the write_nstep0 property for this HistoryVolConfig object"""
+        return self.__write_nstep0
+
+    def set_max_frames(self, nframes, pobj, logger):
+        """Modify the max_frames property of this HistoryVolConfig object.
+        Return True if <nframes> is a valid setting."""
+        nframes_ok = True
+        nframes_i, _ = _is_integer(nframes)
+        nframes_ok = nframes_i and (nframes > 0)
+        if nframes_ok:
+            self.__max_frames = nframes_i
+            self.__max_frames_set = True
+            if logger.getEffectiveLevel() <= logging.DEBUG:
+                ctx = context_string(pobj)
+                logger.debug(f"Setting max frames to '{nframes}'{ctx}")
+            # end if
+        else:
+            emsg = f"Attempt to set max frames to '{nframes}', must be a positive integer"
+            pobj.add_syntax_err(emsg)
+        # end if
+        return nframes_ok
+
+    def set_write_nstep0(self, write_nstep0, pobj, logger):
+        """Modify the write_nstep0 property of this HistoryVolConfig object.
+        Return True if valid"""
+        true_values = ["true", "t", ".true."]
+        false_values = ["false", "f", ".false."]
+        nstep0_ok = True
+        if write_nstep0.lower() in true_values:
+            self.__write_nstep0 = ".true."
+        elif write_nstep0.lower() in false_values:
+            self.__write_nstep0 = ".false."
+        else:
+            nstep0_ok = False
+            emsg = f"Attempt to set write_nstep0 to '{write_nstep0}', must be true or false"
+            pobj.add_syntax_err(emsg)
+        # end if
+        if nstep0_ok and logger.getEffectiveLevel() <= logging.DEBUG:
+            ctx = context_string(pobj)
+            logger.debug(f"Setting write_nstep0 to '{self.__write_nstep0}'{ctx}")
+        # end if
+        return nstep0_ok
+
+    def outfreq_str(self):
+        """Return the output_frequency for this HistoryVolConfig object
+        as a string"""
+        return f"{self.__output_freq[0]}*{self.__output_freq[1]}"
+
+    @property
+    def output_frequency(self):
+        """Return the output_frequency property for this
+        HistoryVolConfig object"""
+        return self.__output_freq
+
+    def set_output_frequency(self, ofreq, pobj, logger):
+        """Modify the output_frequency property of this HistoryVolConfig
+            object. <ofreq> is a tuple consisting of an integer and a period.
+        """
+        if ( isinstance(ofreq, tuple) and (len(ofreq) == 2) and
+             isinstance(ofreq[0], int) and isinstance(ofreq[1], str) and
+             (ofreq[0] > 0) and
+             (ofreq[1].strip().lower() in _TIME_PERIODS)):
```

Comment:
```suggestion
    def __outout_frequency_is_valid(ofreq):
        return isinstance(ofreq, tuple)                   and \
               (len(ofreq) == 2)                          and \
               (ofreq[0] > 0)                             and \
               (ofreq[1].strip().lower() in _TIME_PERIODS)


    def set_output_frequency(self, ofreq, pobj, logger):
        """Modify the output_frequency property of this HistoryVolConfig
            object. <ofreq> is a tuple consisting of an integer and a period.
        """
        if self.__outout_frequency_is_valid(ofreq):
```
Minor change but I get a little worried with multi line if checks.  In this case, I think creating a helper function makes sense because we can give a name to what we are checking.

Reply:
helper function added!